bucket_map/src/restart.rs

```diff
@@ -16,21 +17,27 @@ use {
 const HEADER_VERSION: u64 = 1;
 
 /// written into file at top.
-#[derive(Debug)]
+#[derive(Debug, Pod, Zeroable, Copy, Clone)]
 #[repr(C)]
 pub(crate) struct Header {
     /// version of this file. Differences here indicate the file is not usable.
     version: u64,
     /// number of buckets these files represent.
-    buckets: usize,
+    buckets: u64,
     /// u8 representing how many entries to search for during collisions.
     /// If this is different, then the contents of the index file's contents are likely not as helpful.
     max_search: u8,
     /// padding to get header to u128 aligned
     _dummy: [u8; 15],
 }
 
-#[derive(Debug)]
+// In order to safely guarantee Header is Pod, it cannot have any padding.
+const _: () = assert!(
+    std::mem::size_of::<Header>() == std::mem::size_of::<u128>() * 2,
+    "Header cannot have any padding"
```

Comment:
nit: the error message is not accurate.

Reply:
I suppose it could be more clear if I said no automatic padding can be added to the header. Thoughts?

Reply:
Maybe, just say "incorrect size of header struct" generally.

Reply:
done in next pr